staging/src/k8s.io/kubectl/pkg/cmd/logs/logs.go

```diff
@@ -368,29 +369,34 @@ func (o LogsOptions) RunLogs() error {
 		return err
 	}
 
-	if o.Follow && len(requests) > 1 {
-		if len(requests) > o.MaxFollowConcurrency {
-			return fmt.Errorf(
-				"you are attempting to follow %d log streams, but maximum allowed concurrency is %d, use --max-log-requests to increase the limit",
-				len(requests), o.MaxFollowConcurrency,
-			)
-		}
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	intr := interrupt.New(nil, cancel)
+	return intr.Run(func() error {
+		if o.Follow && len(requests) > 1 {
+			if len(requests) > o.MaxFollowConcurrency {
+				return fmt.Errorf(
+					"you are attempting to follow %d log streams, but maximum allowed concurrency is %d, use --max-log-requests to increase the limit",
+					len(requests), o.MaxFollowConcurrency,
+				)
+			}
```

Comment:
I'd suggest leaving this conditional outside `intr.Run`, inside we should only have the 2 invocations which are expected to be interrupted.

Reply:
Updated. Thank you.